src/renderer/NumberInput.tsx

```diff
@@ -41,31 +74,85 @@ const NumberInput = ({
     }
 
     const newValue: number = parseInt(input, 10);
-    handleChange(newValue);
+    // Prevent user from typing numbers with more than 5 digits
+    if (newValue / 100000 > 1) {
+      return;
+    }
+    setHasChanges(input !== value);
+    setInternalValue(newValue);
+  };
+
+  // Helper for discarding changes
+  const onBlur = () => {
+    setHasChanges(false);
+    setInternalValue(value);
   };
 
-  // Helper for detecting use of the ENTER key
+  const onArrow = async (newValue: number) => {
+    setInternalValue(newValue);
```

Comment:
Make this figure out the new value (i.e. internalValue +/- 1) and do bound checking